src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -353,15 +363,21 @@ TransactionManager::update_lba_mappings(
 
     return lba_manager->update_mappings(
       t, lextents
-    ).si_then([&pextents] {
+    ).si_then([&pextents, this] {
       for (auto &extent : pextents) {
         assert(!extent->is_logical() && extent->is_valid());
         // for non-logical extents, we update its last_committed_crc
         // and in-extent checksum fields
         // For pre-allocated fresh physical extents, update in-extent crc.
-        auto crc = extent->calc_crc32c();
-        extent->set_last_committed_crc(crc);
-        extent->update_in_extent_chksum_field(crc);
+	checksum_t crc;
+	if (extent->get_paddr().is_record_relative() ||
```

Comment:
Need to handle DEVICE_ID_RECORD_RELATIVE case here.

Reply:
I see, looks it is a case not handled correctly yet. The record relative paddr means it will be written inline as the part of the journal. I think this needs a separate decision about whether the device for journaling supports end-to-end protection.